### PR TITLE
[FEATURE] Advanced: Option to skip transaction confirmation (RT-1436)

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

**Comment by rht** [comment]
@yongsoo @annatonger The flag has been split into 3
2015-01-03-213000_1366x768_scrot
@vhpoet Moved settings to user blob (though I think this should be done in a separate PR since other settings are still in local storage).

**Comment by rht** [comment]
@vhpoet Migration to user blob needs a compatibility layer that exports local storage content into user blob if it doesn't exist in the latter. Should I do this altogether here for bridge, blobvault, advanced_feature_switch, websocket server, etc?

**Comment by rht** [comment]
The jade used in the client has bug for mixin argument https://github.com/jadejs/jade/issues/837.
If you put parentheses into one of the argument string, this will lead to syntax error.
So I am forced to use Exchange/Trade instead of Trade (simple) / Trade (advanced).

**Comment by mrajvanshy** [comment]
Is that how we want the UI to look like? 

And I think it's a little confusing to call it exchange and trade settings when those aren't the names of the tabs anymore. Can we not use parentheses and use a dash or something instead?

**Comment by rht** [comment]
@mrajvanshy Exchange/Trade settings name: I will use dash until jade-l10n/jade-l10n-loader dependency has been updated (to 0.35.0 or 1.8.x).

@rippletravis 1. how about "Warning: you won't be notified if your order is far off from the market price". Alternatively, 2. auto-warn the user below the "price of each" input box if the order is far off, much like ripple name input box warning "Recipient should be a Ripple name, a contact, or Ripple/Bitcoin address".

**Comment by rippletravis** [comment]
I think we need to fix the warning in general. If it's the same as an error label, though, I think it could be confusing, as it would feel similar to an invalid input.

I'll let @annatonger craft the text here, but I think that's along the correct lines.

**Comment by annatonger** [comment]
Let's use labels like the ones for Secret key in the Security settings.

Line 1: Send confirmation settings
Line 2:  If you turn this off, you will skip the final confirmation page before you send money. You might pay up to 1% more than you originally specified.

Line 1: Simple trade confirmation settings
Line 2:  If you turn this off, you will skip the final confirmation page before you execute a simple trade. You might pay up to 1% more than you originally specified.

Line 1: Advanced trade confirmation settings
Line 2:  If you turn this off, you will skip the final confirmation page before you place a trade order. 

Thank you @rht !

**Comment by rht** [comment]
2015-01-07-134833_919x634_scrot

**Comment by rht** [comment]
@rippletravis An idea for #2: or use yellow text for the warning + prefix "warning:" (like in mocha test framework, though jshints appears to use the same color for both warnings and errors).

**Comment by rht** [comment]
If the client is to be considered as a "REPL with an enforced structure", then cheap checks should happen early before the command is about to be executed.

**Comment by rippletravis** [comment]
That's a good point. I do think that checks could be better, and happen before confirmation dialogue. I added a task to my backlog in JIRA to work on this. I don't think it should hold up this functionality.

File: src/jade/tabs/advanced.jade [hunk 1]
```diff
@@ -8,6 +8,9 @@ section.col-xs-12.content(ng-controller="AdvancedCtrl")
       .alert.alert-success(ng-show="success.saveBlob", l10n) Your blob vault has been changed successfully.
       .alert.alert-success(ng-show="success.saveBridge", l10n) Your bitcoin bridge has been changed successfully.
       .alert.alert-success(ng-show="success.saveAcctOptions", l10n) Trustline advanced settings has been changed successfully.
+      .alert.alert-success(ng-show="success.saveConfirmation.send", l10n) Send confirmation settings has been changed successfully.
```
Comment (jublo) [inline src/jade/tabs/advanced.jade]: 'settings' is plural. Would the correct form for this and the two following strings not be '[xyz] settings **have** been changed'?

File: src/js/tabs/exchange.js [hunk 2]
```diff
@@ -257,7 +257,11 @@ ExchangeTab.prototype.angular = function (module)
           $scope.confirm_wait = false;
         }, 1000, true);
 
-        $scope.mode = "confirm";
+        if (Options.confirmation.exchange) {
+          $scope.mode = "confirm";
```
Comment (jublo) [inline src/js/tabs/exchange.js]: We prefer single quotes `'` over double ones `"`.

File: src/js/tabs/send.js [hunk 3]
```diff
@@ -881,6 +879,12 @@ SendTab.prototype.angular = function (module)
         'Destination Tag': !!$scope.send.dt,
         'Address': $scope.userBlob.data.account_id
       });
+
+      if (Options.confirmation.send) {
+        $scope.mode = "confirm";
```
Comment (jublo) [inline src/js/tabs/send.js]: See above.

File: src/js/tabs/trade.js [hunk 4]
```diff
@@ -414,6 +412,12 @@ TradeTab.prototype.angular = function(module)
         'Currency pair': $scope.order.currency_pair,
         'Address': $scope.userBlob.data.account_id
       });
+
+      if (Options.confirmation.trade) {
+        $scope.order[type].mode = "confirm";
```
Comment (jublo) [inline src/js/tabs/trade.js]: See above.

**Comment by jublo** [comment]
@rht This needs a rebase.

File: src/jade/tabs/advanced.jade (hunk 1 above)
Comment (rht) [inline src/jade/tabs/advanced.jade]: ...ah, right.

**Comment by rht** [comment]
@jublo rebased and fixed the string quotations in exchange, send, and trade.

**Comment by mrajvanshy** [comment]
It looks like it needs to be rebased again

**Comment by rht** [comment]
Rebased.

**Comment by mrajvanshy** [comment]
LGTM

**Comment by vhpoet** [comment]
LGTM

**Comment by annatonger** [comment]
@rht sorry, can u resolve these conflicts and then we'll merge?

**Pull request merged**